### ISSUE TITLE
AsDataSet() configuration options: Powershell Notation

### ISSUE THREAD

**Issue by jictch** [ExcelDataReader/ExcelDataReader#316]
Your Example for C#:
```
var result = reader.AsDataSet(new ExcelDataSetConfiguration() {
	
        ConfigureDataTable = (_) => new ExcelDataTableConfiguration() {
		    UseHeaderRow = true
	
        }
});
```
My rudimenary Basic snippet in PS5.1:
```
# ExcelDataTableConfiguration => okay
$dTabCnfObj = New-Object ExcelDataReader.ExcelDataTableConfiguration
$dTabCnfObj.UseHeaderRow = $true

# ExcelDataSetConfiguration => okay
$dSetCnfObj = New-Object ExcelDataReader.ExcelDataSetConfiguration
$dSetCnfObj.UseColumnDataType  = $true
# and now beginns the pain (add ExcelDataTableConfiguration to ExcelDataSetConfiguration):
$dSetCnfObj.ConfigureDataTable = "???? [working Example please]"
# $dSetCnfObj.ConfigureDataTable = $dTabCnfOb  didd not work of course ;-)
```
How accomplish the Same functionality in Powershell (native or with linq)?
Can you translate this snippet to Powershell-Natation, please?

Thank you for any assistance

**Comment by andersnm** [comment]
Hi, 

I believe this should work:

```powershell
function ConfigureDataTableCallback($reader) {
	$dTabCnfObj = New-Object ExcelDataReader.ExcelDataTableConfiguration
	$dTabCnfObj.UseHeaderRow = $true
}

$dSetCnfObj = New-Object ExcelDataReader.ExcelDataSetConfiguration
$dSetCnfObj.UseColumnDataType  = $true
$dSetCnfObj.ConfigureDataTable = &ConfigureDataTableCallback

$reader = [ExcelDataReader.ExcelReaderFactory]::CreateReader( ... )
$result = $reader.AsDataSet($dSetCnfObj)

```

The ConfigureDataTable property takes a method which is called for each DataTable

**Comment by jictch** [comment]
Hi andersnm

Thanks for your Suggestion.
Now looks better (no Errors), but the ExcelDataTableConfiguration-Options insyde (in my Case `UseHeaderRow`) takes no Effect on the Result.

Other Idea?

**Comment by andersnm** [comment]
Aplogies, I missed the `return` statement in the callback. Have updated the code sample

**Comment by jictch** [comment]
No Problem. Thanks.
This i have also tested before, but didnt work.

Now throws this error:
```
"ConfigureDataTable": "The value" ExcelDataReader.ExcelDataTableConfiguration "of type
"ExcelDataReader.ExcelDataTableConfiguration" can not be in the type
"System.Func`2 [ExcelDataReader.IExcelDataReader, ExcelDataReader.ExcelDataTableConfiguration]". "
```

**Comment by jictch** [comment]
No i have the Trick found ;-) 

```
$dSetCnfObj = New-Object ExcelDataReader.ExcelDataSetConfiguration
$dSetCnfObj.UseColumnDataType  = $true
$dSetCnfObj.ConfigureDataTable = {
	    $dTabCnfObj = New-Object ExcelDataReader.ExcelDataTableConfiguration
	    $dTabCnfObj.UseHeaderRow = $true
            **return $dTabCnfObj**
}
```

This works for me.

Any better Approaches welcome :-)

**Comment by andersnm** [comment]
That looks neat. I prefer your approach with the anonymous lambda.

Seems I made more mistakes in my code sample, f.ex the syntax for passing a method is `$function:ConfigureDataTableCallback`. Apparently I should stop posting untested code samples for a language I'm not familiar with =)

Have updated my original sample with a more complete script - and actually tested now!

Closing, I think we solved this
